functional_data/lib/src/lenses.dart

```diff
@@ -113,15 +112,15 @@ class List$ {
         },
       );
 
-  static Lens<List<T>, Optional<T>> whereOptional<T>(bool Function(T) predicate) => Lens<List<T>, Optional<T>>(
-        (s) => Optional<T>(s.firstWhereOrNull(predicate)),
+  static Lens<List<T>, T?> whereOptional<T>(bool Function(T) predicate) => Lens<List<T>, T?>(
```

Comment:
even not sure if this function is needed with null-safety. Let me know and I can remove it